docs/client-specifications.rst

```diff
@@ -0,0 +1,246 @@
+.. _client-specifications:
+
+Client Specifications
+=====================
+
+Foreword
+--------
+
+During years, the sole and unique client was Gecko (the platform behind Firefox, Thunderbird..). In order to reach out new platforms and products, a fully-featured Rust client was built, using the application-services components (Viaduct, NSS, ...).
+Unfortunately, this cross-platform was not leveraged by the experimentation team, which built another ad-hoc client. In front of this situation, we are obliged to admit that our idea of having a single client of reference is dead. Instead, we are going to provide specifications for Remote Settings clients, to at least mitigate the consequences of clients fragmentation.
+
+That being said, we still **strongly discourage** the implementation of ad-hoc clients.
+
+We distinguish two major use-cases:
+
+- authenticated write operations, ie. publish data;
+- anonymous read operations, ie. fetch data from within our products.
+
+Since the former does not take place on clients, it matters less than the latter, which has a major impact on traffic and our servers load.
+
+Existing Clients
+----------------
+
++------------------------+-------+-------------+--------------------------------------+---------------+----------+
+|                        | Gecko | Rust Client | application-services/remote-settings | kinto-http.py | kinto.js |
++========================+=======+=============+======================================+===============+==========+
+| Write Operations       | ❌    | ✅          | ❌                                   | ✅            | ✅       |
+| Add/Remove Attachments | ❌    | ✅          | ❌                                   | ❌            | ✅       |
+| Approve/Reject reviews | ❌    | ✅          | ❌                                   | ❌            | ❌       |
++------------------------+-------+-------------+--------------------------------------+---------------+----------+
+| Changesets Endpoints   | ✅    | ✅          | ❌                                   | ✅            | ❌       |
+| Cache Busting          | ✅    | ✅          | ❌                                   | ✅            | ❌       |
+| Env Switching          | ✅    | ~ Partial   | ✅                                   | ❌            | ❌       |
+| Signature Verification | ✅    | ✅ Optional | ❌                                   | ❌            | ❌       |
+| Local State            | ✅    | ✅ Optional | ❌                                   | ❌            | ✅       |
+| Fetch Attachments      | ✅    | ✅          | ✅ Without integrity check           | ❌            | ✅       |
+| Backoff                | ✅    | ✅          | ✅                                   | ✅            | ✅       |
+| Deprecation            | ❌    | ❌          | ❌                                   | ✅ Minimal    | ✅       |
++------------------------+-------+-------------+--------------------------------------+---------------+----------+
```

Comment:
Don't know if it's just Github, but this table is rendering a little funky <img width="928" alt="image" src="https://github.com/mozilla/remote-settings/assets/20747904/62dcbe39-0cd0-4eda-8294-82a169283b02">